Source/Core/Cartesian3.js

```diff
@@ -58,9 +60,7 @@ define([
      */
     Cartesian3.fromSpherical = function(spherical, result) {
         //>>includeStart('debug', pragmas.debug);
-        if (!defined(spherical)) {
-            throw new DeveloperError('spherical is required');
-        }
+        check.type.object(spherical);
```

Comment:
I have profiled this, though I should probably run this longer and with a few different scenarios to really see how it works out.

After running Cesium for a short while (with terrain) check.type.object is at 0.00% and check.defined is at 0.21%.